### ISSUE TITLE
pysam.depth is not fully consistent with samtools depth v1.11+

### ISSUE THREAD

**Issue by masikol** [pysam-developers/pysam#1018]
Hello there!

Just noticed an inconsistency of `pysam.depth` (pysam version `0.16.0.1`) with `samtools depth` v1.11+. The thing is that `pysam.depth` does not support option `-J`, although this option was introduced in `samtools` [v1.11](https://github.com/samtools/samtools/releases/tag/1.11), and [README](https://github.com/pysam-developers/pysam/blob/master/README.rst) states that "The current version of pysam wraps 3rd-party code from htslib-1.11, samtools-1.11, and bcftools-1.11".

I detected this issue by running the following command:
```
pysam.depth('-a', '-J', '--reference', '/mnt/1.5_drive_0/kromsatel-dev/reference/Wuhan-Hu-1-compele-genome.fasta', '-o', 'depth.out.txt', 'Wuhan-Hu-1_cleaned.sorted.bam')
```
It resulted in following:
```
depth: invalid option -- 'J'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/cager/Misc_soft/py-venv/lib/python3.8/site-packages/pysam/utils.py", line 69, in __call__
    raise SamtoolsError(
pysam.utils.SamtoolsError: 'samtools returned with error 1: stdout=, stderr=\nUsage: samtools depth [options] in1.bam [in2.bam [...]]\nOptions:\n   -a                  output all positions (including zero depth)\n   -a -a (or -aa)      output absolutely all positions, including unused ref. sequences\n   -b <bed>            list of positions or regions\n   -X                  use customized index files\n   -f <list>           list of input BAM filenames, one per line [null]\n   -H                  print a file header\n   -l <int>            read length threshold (ignore reads shorter than <int>) [0]\n   -d/-m <int>         maximum coverage depth [8000]. If 0, depth is set to the maximum\n                       integer value, effectively removing any depth limit.\n   -o FILE             where to write output to [samtools_stdout]\n   -q <int>            base quality threshold [0]\n   -Q <int>            mapping quality threshold [0]\n   -r <chr:from-to>    region\n   -g <flags>          include reads that have any of the specified flags set [0]\n   -G <flags>          filter out reads that have any of the specified flags set                       [UNMAP,SECONDARY,QCFAIL,DUP]\n      --input-fmt-option OPT[=VAL]\n               Specify a single input file format option in the form\n               of OPTION or OPTION=VALUE\n      --reference FILE\n               Reference sequence FASTA FILE [null]\n      --verbosity INT\n               Set level of verbosity\n\nThe output is a simple tab-separated table with three columns: reference name,\nposition, and coverage depth.  Note that positions with zero coverage may be\nomitted by default; see the -a option.\n\n'
```
I see that "Usage" string in the error message does not mention option `-J`. So it is by no means an error. But it would be great if this option was supported in pysam.

Thanks in advance!

**Comment by jmarshall** [comment]
README on the **master** branch says that pysam there wraps samtools-1.11 etc. 1.11 was merged in PR #962 in January.

However the **0.16.0.1 release** is from June 2020 and still wraps samtools-1.10 etc. So `pysam.depth()` in that release does not have this `-J` option.

So you could build current pysam **master** yourself to get the updated `pysam.depth()` or wait for the next pysam release — although at this point I want to update it again to wrap 1.12 first.

**Comment by masikol** [comment]
Alright. Thank you for the explanation!
